### PR TITLE
[pvr.tvh] link events with recordings and vice versa

### PR DESCRIPTION
**Pull request by Jalle19**
See https://github.com/xbmc/xbmc/pull/3900 for what it's supposed to do. I haven't tested this yet.

### CODE REVIEW

File: addons/pvr.tvh/src/Tvheadend.cpp [hunk 1]
```diff
@@ -626,6 +627,10 @@ void CTvheadend::TransferEvent
   epg.iEpisodeNumber      = event.episode;
   epg.iEpisodePartNumber  = event.part;
 
+  std::stringstream ss;
+  ss << event.recordingId;
+  epg.strRecordingId = ss.str().c_str();
```
Comment (adamsutton) [inline addons/pvr.tvh/src/Tvheadend.cpp]: Surely that doesn't work?

Comment (Jalle19) [inline addons/pvr.tvh/src/Tvheadend.cpp]: It's supposed to. You can throw almost anything at a string stream.

Comment (Jalle19) [inline addons/pvr.tvh/src/Tvheadend.cpp]: Okay, so I guess what's really needed is something like this (ignoring the correct strcpy() syntax):

```
const std::string recordingId = ss.str();
strcpy(epg.strRecordingId, recordingId.c_str())
```

**Comment by Jalle19** [comment]
@adamsutton should this work? It seems strncpy() can't be used because it takes char\* as the destination, not const char*.

**Pull request merged**